fix: parse label for dataset settings for text and token classification tasks (#3497)
recognai/rubrix: CHANGELOG.md

## [Unreleased]

### Fixed

> - Dataset settings (text and token classification) are properly parsed to strings for clients and for the backend endpoint (Closes [3495](https://github.com/argilla-io/argilla/issues/3495)).

Reviewer: ```suggestion
  - `TextClassificationSettings` and `TokenClassificationSettings` labels are properly parsed to strings both in the Python client and in the backend endpoint (Closes [#3495](https://github.com/argilla-io/argilla/issues/3495)).
```
